src/storage-client/src/controller.rs

```diff
@@ -1785,9 +1951,109 @@ where
         Ok(())
     }
 
-    /// Acquires persist handles for the given `shard`. This will `halt!` the
-    /// process if we cannot successfully acquire a critical handle with our
-    /// current epoch.
+    /// Returns IDs for all storage collections that the given
+    /// `collection_description` depends on.
+    ///
+    /// TODO: @sean: This is where the remap shard would slot it.
+    fn get_collection_dependencies(
+        &self,
+        collection_description: &CollectionDescription<T>,
+    ) -> Vec<GlobalId> {
+        let mut result = Vec::new();
+
+        if let DataSource::Ingestion(ingestion) = &collection_description.data_source {
+            match &ingestion.desc.envelope {
+                SourceEnvelope::Debezium(envelope_debezium) => {
+                    let tx_metadata_topic = envelope_debezium.dedup.tx_metadata.clone();
```

Comment:
You should be able to avoid the clone: `let tx_metadata_topic = &envelope_debezium.dedup.tx_metadata;`